kylin-spark-project/kylin-spark-metadata/src/main/scala/org/apache/kylin/engine/spark/metadata/MetadataConverter.scala

```diff
@@ -115,6 +117,21 @@ object MetadataConverter {
     dimensionIndex
   }
 
+  def extractAllRowkeyColumnDesc(cubeInstance: CubeInstance): java.util.LinkedHashMap[Integer, ColumnDesc] = {
```

Comment:
There are some lines of code are duplicated in `extractAllRowkeyColumnDesc` `extractAllColumnDesc`. Maybe we can improve the code smell by merge them into one method?